### PR TITLE
[NO_REVIEW] JAMES-3477 demonstrates concurrency issue in MimeMessageCOW

### PR DESCRIPTION
**Pull request by chibenwa**
CI for https://github.com/apache/james-project/pull/280

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
```
[dbb98050fece70d2bdf8566cc3553946546e6739] [ERROR] /james-project/server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/MessageSender.java:[79,11] exception javax.mail.MessagingException is never thrown in body of corresponding try statement
```

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/MimeMessageBenchTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.Stopwatch;
+
+/**
+ * This benches the efficiency at which James mailetcontainer splits emails.
+ */
+@Disabled
+public class MimeMessageBenchTest {
+    private static String CONTENT = "0123456789\r\n".repeat(1024 * 10); // 120KB message
```
Comment (rouazana) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/MimeMessageBenchTest.java:60]: what happens with biggest messages?
is there significant differences in term of memory usage?
in term of GC?

Comment (chibenwa) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/MimeMessageBenchTest.java:60]: That's for sure, more than this bench would be needed to adopt such a drastic change.

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/MimeMessageBenchTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import org.junit.rules.TemporaryFolder;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.Stopwatch;
+
+/**
+ * This benches the efficiency at which James mailetcontainer splits emails.
+ */
+@Disabled
+public class MimeMessageBenchTest {
```
Comment (Arsnael) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/MimeMessageBenchTest.java:59]: I guess you probably tried to set that up quickly for testing, but we should try to make the effort of writing new code with JUnit5 I think... It counters the efforts made sometimes to reduce the usage of JUnit4 in our tests

Comment (chibenwa) [reply server/mailet/integration-testing/src/test/java/org/apache/james/mailets/MimeMessageBenchTest.java:59]: Not really

All mailet integration tests are written in JUNIT 4

Writing a single one in JUNIT 5 is a huge effort. Let's not link things together with uneeded requirements, overwize nothing will ever happen.

Also, that's an experiment, in a draft pull request, POCing around what's possible to do, at which costs on a complicated topic. My goal here is to gain enough insight on the MimeMessage topic to have a lightened discussion later on on the ML about it. Not get a merge (hence draft PR status)

**Comment by chibenwa** [comment]
JamesDeploymentValidationTest>DeploymentValidation.validateDeployment:81 » InvalidServerResponse

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
See https://github.com/apache/james-project/pull/282

**Pull request closed without merging**